modules/install/join_k3s_agent.sh

```diff
@@ -35,65 +65,47 @@ add_config() {
 }
 
 rhel() {
-   local node_os="${1}"
-   local username="${2}"
-   local password="${3}"
-
   if [ "$node_os" = "rhel" ]
-    then
-      subscription-manager register --auto-attach --username="$username" --password="$password"
-      subscription-manager repos --enable=rhel-7-server-extras-rpms
+  then
+    subscription-manager register --auto-attach --username="$rhel_username" --password="$rhel_password"
+    subscription-manager repos --enable=rhel-7-server-extras-rpms
   fi
 }
 
 disable_cloud_setup() {
-   local node_os="${1}"
-
-if  [[ "$node_os" = *"rhel"* ]] || [[ "$node_os" = *"centos"* ]]
+  if  [[ "$node_os" = *"rhel"* ]] || [[ "$node_os" = *"centos"* ]]
```

Comment:
Is this something we want to do regardless of node_os? I don't think it needs to change as part of this PR as this is just a refactor to what is existing, but thought it might be a good discussion point...

Reply:
Agrees, lets discuss and I can add that change

Reply:
We discussed internally and decided we might add to a future PR, but not necessary as part of this one.